### ISSUE TITLE
Axe-core in NextJS 14 with app router

### ISSUE THREAD

**Issue by JackGasoline** [larsmagnus/learn#1]
Hello Lars! I am just starting my journey in the world of accessibility and I was very happy to find your article on axe-core in React and NextJS. I was wondering, how would you go about implementing it in a NextJS 14 setup with app router? Where would you include the reportAccessibilty component in such a scenario?
I have tried some things but it does not seem to work… It would be great if you could help me out with a basic example. Thank you!

**Comment by larsmagnus** [comment]
Hey Jack! That's awesome that you're starting your accessibility journey! Welcome 👋 

Thanks for checking out my article, and also for reaching out. Solving this in Next 14 is a bit different, and there are multiple ways this can be done. 

- I've set up a full example here: https://github.com/larsmagnus/learn/tree/main/axe-core/create-next-app-next14
- This new `create-next-app-next14` folder contains a full example of how to report accessibility violations in Next 14. 
- Check out [`app/layout.tsx`](https://github.com/larsmagnus/learn/blob/main/axe-core/create-next-app-next14/app/layout.tsx) to see where you could include it. 
- To not cause issues on the server, I've implemented the reporter as a client component provider. You can read more about client/server composition patterns in Next.js here https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns

Again, this is just a single example, and there are several other possible ways to solve this. So I hope this works for your use case 🙏 

Let me know if you have any more questions!

**Comment by JackGasoline** [comment]
Thank you for the quick reply and the example! This works very well for my project! 🙏🙏🙏

**Comment by larsmagnus** [comment]
You're welcome @JackGasoline. Good luck with your project! 🚀
